backend/btrixcloud/operator.py

```diff
@@ -848,30 +909,49 @@ def sync_pod_status(self, pods, status):
                     ):
                         running = True
 
-                    terminated = cstatus["state"].get("terminated")
-                    exit_code = terminated and terminated.get("exitCode")
-                    if terminated and exit_code:
-                        crash_time = terminated.get("finishedAt")
-                        pod_status = status.podStatus[name]
-                        pod_status.isNewCrash = pod_status.crashTime != crash_time
-                        pod_status.crashTime = crash_time
-
-                        # detect reason
-                        if terminated.get("reason") == "OOMKilled" or exit_code == 137:
-                            pod_status.reason = "oom"
-                        else:
-                            pod_status.reason = "interrupt: " + str(exit_code)
+                    self.handle_terminated_pod(
+                        name, role, status, cstatus["state"].get("terminated")
+                    )
 
                 if role == "crawler":
                     crawler_running = crawler_running or running
+                    done = done and phase == "Succeeded"
                 elif role == "redis":
                     redis_running = redis_running or running
 
         # pylint: disable=broad-except
         except Exception as exc:
+            done = False
             print(exc)
 
-        return crawler_running, redis_running
+        return crawler_running, redis_running, done
+
+    def handle_terminated_pod(self, name, role, status, terminated):
+        """handle terminated pod state"""
+        if not terminated:
+            return
+
+        exit_time = terminated.get("finishedAt")
+        if not exit_time:
+            return
```

Comment:
If the pod is terminated and we don't have a `finishedAt` time for some reason, should we fall back to `datetime.now()`? Or is this preventing multiple calls?

Reply:
Hm, I think that shouldn't really happen at all, as startedAt and finishedAt are required fields, according to:
https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.28/#containerstateterminated-v1-core

Maybe should remove this check altogether (not checking if startedAt is set)
If it is empty somehow and then set, could result in double-counting as we check to see if its a new exit based on value of that field.

For now, added a print warning to indicate if this ever happens..